File: backend/src/main/java/br/com/rafaelmaia/mar_de_beleza_system/domain/entity/Payment.java
package br.com.rafaelmaia.mar_de_beleza_system.domain.entity;

import br.com.rafaelmaia.mar_de_beleza_system.domain.enums.PaymentMethod;
import br.com.rafaelmaia.mar_de_beleza_system.domain.enums.PaymentStatus;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "tb_payment")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @EqualsAndHashCode.Include
    private Long id;

    @OneToOne
    @JoinColumn(name = "appointment_id", nullable = false, unique = true)
    private Appointment appointment;

    @Column(nullable = false)
    private BigDecimal totalAmount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentMethod paymentMethod;

    @Column(nullable = false)
    private LocalDateTime paymentDate;

    private String observations;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private PaymentStatus status;
}

File: frontend/src/types/payment.ts
import type { Appointment } from './appointment';

export type PaymentMethod = 'PIX' | 'CREDIT_CARD' | 'DEBIT_CARD' | 'CASH';
export type PaymentStatus = 'PAID' | 'CANCELED';

export type Payment = {
  id: number;
  appointment: Appointment;
  totalAmount: number;
  paymentDate: string; // ex: "2025-07-30T21:45:00"
  paymentMethod: PaymentMethod;
  status: PaymentStatus;
  observations?: string;
};